pkg/stub/imagestreams.go

```diff
@@ -208,8 +208,18 @@ func (h *Handler) upsertImageStream(imagestreamInOperatorImage, imagestreamInClu
 		return nil
 	}
 
-	imagestreamInOperatorImage.ResourceVersion = imagestreamInCluster.ResourceVersion
-	_, err := h.imageclientwrapper.Update(imagestreamInOperatorImage)
+	// with the advent of removing EOL images in 4.2, we cannot just update the imagestream
+	// with client-go/Update as that will replace vs. patch; we need to preserve EOL specs
+	// if the imagestream was created during 4.1 as to allow pullthrough on builds for example
+	// so we manually merge the tags from the payload onto the existing image
+	for _, newTag := range imagestreamInOperatorImage.Spec.Tags {
+		for i, oldTag := range imagestreamInCluster.Spec.Tags {
+			if newTag.Name == oldTag.Name {
+				imagestreamInCluster.Spec.Tags[i] = newTag
+			}
+		}
+	}
+	_, err := h.imageclientwrapper.Update(imagestreamInCluster)
```

Comment:
it looks like this means you're only updating the spec.tags in the imagestream.  There could be top level imagestream fields/metadata that also changed, right?

Reply:
good point ... will update

Reply:
there is also a problem with this logic in while EOL ones are getting preserved, new ISTs are not getting added

Reply:
so as part of the fix I've for adding new tags I've switch back to doing the Update with the imagestream from the operator payload, but where that imagestream has been updated to include missing tags from the older version.

Reply:
that sounds like a sensible approach...  essentially making the imagestream update purely additive from a tag perspective.